src/pyscheduling/JS/JobShop.py

```diff
@@ -10,22 +10,214 @@
 import warnings
 
 import numpy as np
+import networkx as nx
 import matplotlib.pyplot as plt
 
 import pyscheduling.Problem as RootProblem
 import pyscheduling.SMSP.riPrecLmax as riPrecLmax
+import pyscheduling.SMSP.rihiCi as rihiCi
+from pyscheduling.Problem import GenerationLaw, Job
 
+class GenerationProtocol(Enum):
+    BASE = 1
 
-Job = namedtuple('Job', ['id', 'start_time', 'end_time'])
 
-class GenerationProtocol(Enum):
-    VALLADA = 1
+Node = namedtuple('Node', ['machine_id', 'job_id'])
+
+@dataclass
+class JobsGraph:
+    instance: RootProblem.Instance
```

Comment:
Why do you need the instance here ?

For me the graph is an attribute in the solution, which has already the reference to instance. I think we should keep the graph as attribute to avoid implementing stuff that are dependent on the solution outside of it.